app/posts/finishing-touches-for-forms.md

```diff
@@ -0,0 +1,149 @@
+
+There are, however, some suggestions in the Design System guidance that go beyond the remit of the form builder.
+
+They are often forgotten, we'll go over some simple ways to solve them here.
+
+## Positioning the error summary at the top of the page
+
+In addition to making it clear that something is wrong, we also need to tell the user exactly what happened and how to fix it.
+
+[The guidance says](https://design-system.service.gov.uk/components/error-summary#where-to-put-the-error-summary):
+
+> Put the error summary at the top of the main container. If your page includes breadcrumbs or a back link, place it below these, but above the `<h1>`.
+
+This is problematic because the form often isn't the first thing on the page but the form builder needs to be responsible for rendering the error summary to ensure the links and targets are consistent.
+
+We could solve the problem by wrapping the whole page in a `<form>` element, which would allow us to place the error summary wherever we like, but it's a hack. It isn't semantically correct and is likely to lead to other problems.
+
+Thankfully, Rails has our backs. We can use [`content_for`](https://guides.rubyonrails.org/layouts_and_rendering.html#using-the-content-for-method) to insert content into a named block in our layout. If we add a named block called `top_of_main` at the top of our `<main>` element, like this:
+
+```html
+<div class="govuk-width-container">
+  <main class="govuk-main-wrapper" id="main-content">
+    <%= yield :top_of_main %>
+    <h1 class="govuk-heading-xl">Default page template</h1>
+
+    <!-- the rest of our page -->
+```
+
+We can send our error summary from the form to it, like this:
+
+```html
+<%= form_with(model: @object, url: some_path) do |form| %>
+  <%= content_for(:top_of_main) { form.govuk_error_summary } %>
+
+  <%= form.govuk_text_field(:how_many_juggling_items) %>
+<% end %>
+```
+
+And now our error summary is in the right spot.
+
+![Two screenshots side by side showing a form with errors. The left one has the error summary after the h1 and the right one has the error summary first](/assets/posts/finishing-touches-for-forms/error-summary-side-by-side.png)
+
+## Prefixing the page title with 'Error:'
+
+When a form submission results in a validation failure we need to make it clear to the user that something is wrong.
+
+[The guidance says](https://design-system.service.gov.uk/patterns/validation/#how-to-tell-the-user-about-validation-errors):
+
+> Add ‘Error: ’ to the beginning of the page `<title>` so screen readers read it out as soon as possible
+
+This is a little tricky because we need some logic that can determine whether there's an error on the page, and we set the `<title>` in the document `<head>` before we've rendered the form in the `<body>`.
+
+We can use `content_for` to our advantage here too to place the title where we need it.
+
+```html
+<head>
+  <%= tag.title(yield(:page_title)) %>
+  <!-- other head content -->
+</head>
+```
```

Comment:
Now this section has been rewritten to mention the helper from GOV.UK Components, I’m a little unsure how this example relates to the helper below. Are they related? Is this trying to demonstrate the old and new, helper-based approach?

Reply:
I've reworded this. I think the 'Now we' helps here because it's continuing from the step above.